JamGame/Assets/Scripts/TileUnion/PlaceCondition.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using TileBuilder;
+using TileUnion.Tile;
+using UnityEngine;
+
+namespace TileUnion.PlaceCondition
+{
+    [InterfaceEditor]
+    public interface IPlaceCondition
+    {
+        public Result<List<TileImpl>> ApplyCondition(
+            TileUnionImpl targetTileUnion,
+            TileBuilderImpl tileBuilderImpl
+        );
```

Comment:
This signature implies that `FailResult `is success and `SuccessResult` is fail what's very confusing